tdwii_plus_examples/handlers.py

```diff
@@ -357,18 +357,28 @@ def handle_nget(event, db_path, cli_config, logger):
     addr, port = requestor.address, requestor.port
     logger.info(f"Received N-GET request from {addr}:{port} at {timestamp}")
 
-    model = event.request.AffectedSOPClassUID
+    model = event.request.RequestedSOPClassUID
+    if len(model) == 0:
+        print("Requested SOP Class UID not provided")
+    # need to manipulate the content of the request
+    # also need to validate the content and reject if it's not DICOM conformant and IHE-RO
+    # TDW-II profile adherent
+    ds_from_request = Dataset()
+    # decode(event.request.ModificationList, True, True)
+    ds_from_request.SOPInstanceUID = event.request.RequestedSOPInstanceUID
+    if ds_from_request.SOPInstanceUID is None or len(ds_from_request.SOPInstanceUID) == 0:
+        print("Requested SOP Instance UID is empty")
 
     engine = create_engine(db_path)
     with engine.connect() as conn:  # noqa:  F841
         Session = sessionmaker(bind=engine)
         session = Session()
         # Search database using Identifier as the query
         try:
-            matches = search(model, event.identifier, session)
+            matches = search(model, ds_from_request, session)
```

Comment:
at line 378, I wonder how ds_from_request initialized in this method can contain the identifier. I do not see the event.identifier has been added to ds_from_request

Reply:
event.identifier is not populated by pynetdicom for N-GET.  Which is why it was failing.  
A few lines higher up is
`ds_from_request.SOPInstanceUID = event.request.RequestedSOPInstanceUID`
is how the required elements to match is provided (the Requested SOP Instance UID in the N-GET-RQ is used to look up the UPS SOP Instance UID)